src/optimisations/keyset-deduplication.js

```diff
@@ -17,10 +17,20 @@ function isSortedArrayOfThingsSameAsSortedArrayOfThings(a, b) {
 export default class KeysetDeduplicationOptimisation implements Extension<{}, Array<any>, Array<Keyset>> {
   keysets : Array<Keyset>
   frequencies : Array<number>
+  omittedKeysetCount: number
 
-  constructor() {
-    this.keysets = [];
+  constructor({ omittedKeysets = [] } : { omittedKeysets : Array<Keyset>} = {}) {
+    this.keysets = omittedKeysets.slice(0);
```

Comment:
I generally prefer `[].slice.call(array)` for copying. On modern browsers, `[...array]` is even better (though we shouldn't do that here).